Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp

```diff
@@ -0,0 +1,479 @@
+    // unequal volumes always means separate hierarchies
+    if (ref.volume() != cmp.volume()) {
+      return false;
+    }
+    // within the same volume hierarchy only consider layers
+    return (ref.layer() == cmp.layer());
+  };
+  auto geoSelBeg = m_cfg.geometrySelection.begin();
+  auto geoSelEnd = m_cfg.geometrySelection.end();
+  // sort geometry selection so the unique filtering works
+  std::sort(geoSelBeg, geoSelEnd);
+  auto geoSelLastUnique = std::unique(geoSelBeg, geoSelEnd, isDuplicate);
+  if (geoSelLastUnique != geoSelEnd) {
+     ACTS_WARNING("Removed " << std::distance(geoSelLastUnique, geoSelEnd)
+                             << " geometry selection duplicates");
+    m_cfg.geometrySelection.erase(geoSelLastUnique, geoSelEnd);
+  }
+  ACTS_INFO("Hough geometry selection:");
+  for (const auto& geoId : m_cfg.geometrySelection) {
+    ACTS_INFO("  " << geoId);
+  }
+
+  // Fill convenience variables
+  m_step_x = (m_cfg.xMax - m_cfg.xMin) / m_cfg.houghHistSize_x;
+  m_step_y = (m_cfg.yMax - m_cfg.yMin) / m_cfg.houghHistSize_y;
+  for (unsigned i = 0; i <= m_cfg.houghHistSize_x; i++)
+    m_bins_x.push_back(
+        unquant(m_cfg.xMin, m_cfg.xMax, m_cfg.houghHistSize_x, i));
+  for (unsigned i = 0; i <= m_cfg.houghHistSize_y; i++)
+    m_bins_y.push_back(
+        unquant(m_cfg.yMin, m_cfg.yMax, m_cfg.houghHistSize_y, i));
+}
+
+ActsExamples::ProcessCode ActsExamples::HoughTransformSeeder::execute(
+    const AlgorithmContext& ctx) const {
+
+   // clear our Hough measurements out from the previous iteration, if at all
+   houghMeasurementStructs.clear();
+  
+   // add SPs to the inputs
+   AddSPs(ctx); 
+
+   // add ACTS measurements
+   AddMeasurements(ctx);
+    
+   static thread_local ProtoTrackContainer protoTracks;
+   protoTracks.clear();
+
+   // loop over our subregions and run the Hough Transform on each
+  for (auto subregion : m_cfg.subRegions) {
+    ActsExamples::HoughHist m_houghHist =
+        createHoughHist(subregion);
+
+    for (unsigned y = 0; y < m_cfg.houghHistSize_y; y++)
+      for (unsigned x = 0; x < m_cfg.houghHistSize_x; x++)
+        if (passThreshold(m_houghHist, x, y)) {
+          /* now we need to unpack the hits; there should be multiple track
+             candidates if we have multiple hits in a given layer So the first
+             thing is to unpack the indices (which is what we need) by layer */
+          std::vector<std::vector<Index>> hitIndicesAll(
+              m_cfg.nLayers);  // [layer,Index]
+          std::vector<size_t> nHitsPerLayer(m_cfg.nLayers);
+          for (auto measurementIndex : m_houghHist(y, x).second) {
+             HoughMeasurementStruct *meas = houghMeasurementStructs[measurementIndex].get();
+             hitIndicesAll[meas->layer].push_back(meas->index);
+             nHitsPerLayer[meas->layer]++;
+          }
+
+          std::vector<std::vector<int>> combs = getComboIndices(nHitsPerLayer);
+
+          for (size_t icomb = 0; icomb < combs.size();
```

Comment:
We have a utility class `Acts/Utilities/Enumerate.hpp` which allows you to do 

```c++
for (auto [icomb, comb] : enumerate(combs)){
 
std::cout << "Index " << comb << " has value " << comb << std::end;
}
```

Reply:
1) For the B field automatic generation, that sounds nice. Is it necessarily something that works in all cases? For more complex geometries with B field corrections is it clear that it's easy to get the constant value? (Maybe, I have not thought it through), and is there an example of where I can access the constant B field (tried looking and not obvious)
2) I tried asking some months ago about the grid class. I see the readthedocs instructions, but I haven't seen an obvious, simple explanation of how this should be used (I was pointed to SpacePointGrid.ipp but... well, it isn't so helpful to me, maybe that's my fault)